### PR TITLE
EO-812: Added injections column to health score overview

### PR DESCRIPTION
**Pull request by PatrickSison**
### What Changed
 - Added injections column to health score page

### How To Test
 - [Point openresty to prd](https://confluence.int.messagesystems.com/pages/viewpage.action?spaceKey=ENG&title=Configuring+Openresty+Upstreams)
- Check the health score page and see the new column after `Current Score`
 - Select a custom date before 7 Jun 2019 (new model date)
 - Column should not be sortable and it should render `- - -` instead

### To Do
- [ ] Address feedback

### CODE REVIEW

File: src/pages/signals/components/HealthScoreOverview.js [hunk 1]
```diff
@@ -94,14 +95,15 @@ class HealthScoreOverview extends React.Component {
...
-      data, error, facet, loading, signalOptions, subaccounts, tableName, totalCount, header
+      data, error, facet, loading, signalOptions, subaccounts, tableName, totalCount, header, to
     } = this.props;
 
     const subaccountFilter = _.get(signalOptions, 'subaccount.id');
     const isCustomRange = signalOptions.relativeRange === 'custom';
 
     const noFacetSelected = facet.key === 'sid';
     const noSubaccountFilter = subaccountFilter === undefined;
+    const afterNewModel = moment('2019/06/7').diff(to) <= 0;
```
Comment (rajumsys) [inline src/pages/signals/components/HealthScoreOverview.js]: I guess this is why getting the following console warning
```
console.js:39 Deprecation warning: value provided is not in a recognized RFC2822 or ISO format. moment construction falls back to js Date(), which is not reliable across all browsers and versions. Non RFC2822/ISO date formats are discouraged and will be removed in an upcoming major release. Please refer to http://momentjs.com/guides/#/warnings/js-date/ for more info.
Arguments: 
[0] _isAMomentObject: true, _isUTC: false, _useUTC: false, _l: undefined, _i: 2019/06/7, _f: undefined, _strict: undefined, _locale: [object Object]
Error
    at Function.createFromInputFallback (http://localhost:3100/static/js/7.chunk.js:118714:89)
    at configFromString (http://localhost:3100/static/js/7.chunk.js:120776:11)```

Comment (rajumsys) [reply src/pages/signals/components/HealthScoreOverview.js]: Can we use a valid date (for moment) and also put it in a constant somewhere (this module is fine too).

File: src/pages/signals/components/HealthScoreOverview.js [hunk 2]
```diff
@@ -180,11 +182,21 @@ class HealthScoreOverview extends React.Component {
               <NumericDataCell value={current_health_score} />
             )}
           />
+          <Column
+            align="right"
+            dataKey="current_total_injection_count"
+            label={isCustomRange ? 'Injections' : 'Current Injections'}
```
Comment (rajumsys) [inline src/pages/signals/components/HealthScoreOverview.js]: This is probably fine but just making sure, ticket is saying it should be just `Current Injections`?

File: src/pages/signals/components/HealthScoreOverview.js [hunk 3]
```diff
@@ -180,11 +182,21 @@ class HealthScoreOverview extends React.Component {
...
+            sortable={afterNewModel}
+            width="15%"
+            component={({ current_total_injection_count }) => (
+              <NumericDataCell value={afterNewModel ? current_total_injection_count : null} />
```
Comment (rajumsys) [inline src/pages/signals/components/HealthScoreOverview.js]: Should we hide this column completely in that case? I mean we're not communicating in UI that it's only viewable in certain condition. So people might be wondering why it's not showing any value. But if we hide, they will know at least the column is not their (but i guess it'll still be a question).

File: src/pages/signals/components/tests/HealthScoreOverview.test.js [hunk 4]
```diff
@@ -93,6 +94,12 @@ describe('HealthScoreOverview', () => {
     expect(wrapper.find('Column[dataKey="current_health_score"]').prop('label')).toEqual('Score');
   });
 
+  it('render injections properly when before new model date', () => {
+    const wrapper = subject({ signalOptions: { relativeRange: 'custom' }, to: '2019/06/05' });
+    expect(wrapper.find('Column[dataKey="current_total_injection_count"]').prop('label')).toEqual('Injections');
+    expect(wrapper.find('Column[dataKey="current_total_injection_count"]').prop('sortable')).toEqual(false);
```
Comment (rajumsys) [inline src/pages/signals/components/tests/HealthScoreOverview.test.js]: how about a test showing where `sortable` will be `true`?

Comment (PatrickSison) [reply src/pages/signals/components/tests/HealthScoreOverview.test.js]: I was going based off of similarly the Current Score vs Score which is based on the selected date. I'll ask Avi in the morning

File: src/pages/signals/components/HealthScoreOverview.js (hunk 3 above)
Comment (PatrickSison) [reply src/pages/signals/components/HealthScoreOverview.js]: I was going based off of similarly the Current Score vs Score which is based on the selected date. I'll ask Avi in the morning

File: src/pages/signals/components/HealthScoreOverview.js (hunk 2 above)
Comment (PatrickSison) [reply src/pages/signals/components/HealthScoreOverview.js]: I was going based off of similarly the Current Score vs Score which is based on the selected date. I'll ask Avi in the morning

File: src/pages/signals/components/HealthScoreOverview.js (hunk 3 above)
Comment (PatrickSison) [reply src/pages/signals/components/HealthScoreOverview.js]: Only issue with this is the width will have to go to another column or multiple. Currently there's no autowidth, so the same condition would need to apply to 1 or several more columns

Comment (PatrickSison) [reply src/pages/signals/components/HealthScoreOverview.js]: I'll ask about this one as well

File: src/pages/signals/components/HealthScoreOverview.js (hunk 2 above)
Comment (PatrickSison) [reply src/pages/signals/components/HealthScoreOverview.js]: We're going ahead with `Injections` if it's a custom range and `Current Injections` if it's the most recent data

File: src/pages/signals/components/tests/HealthScoreOverview.test.js [hunk 5]
```diff
@@ -94,19 +94,11 @@ describe('HealthScoreOverview', () => {
...
-    expect(wrapper.find('Column[dataKey="current_total_injection_count"]').prop('sortable')).toEqual(false);
-  });
-
-  it('render injections properly when after model date', () => {
-    const wrapper = subject({ signalOptions: { relativeRange: 'custom' }, to: new Date('2019/06/07') });
-    expect(wrapper.find('Column[dataKey="current_total_injection_count"]').prop('label')).toEqual('Injections');
-    expect(wrapper.find('Column[dataKey="current_total_injection_count"]').prop('sortable')).toEqual(true);
+  it('renders injections column if after V2 release date', () => {
+    const wrapper = subject({ signalOptions: { relativeRange: 'custom', to: new Date('2019/06/08') }});
+    expect(wrapper).toMatchSnapshot();
```
Comment (rajumsys) [inline src/pages/signals/components/tests/HealthScoreOverview.test.js]: Previous inline explicit assertion was better but I can see why you did snapshot though. 

We need a test for `Current Injection` code path.

**Pull request closed without merging**